src/App.tsx

```diff
@@ -1,9 +1,22 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
+let visibleMovies: Movie[] = [];
+
 export const App: React.FC = () => {
+  const [search, setSearch] = useState('');
+
+  const searchHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const lowerCaseEvent = event.target.value.toLowerCase();
+
+    setSearch(event.target.value);
+    visibleMovies = moviesFromServer.filter(element => (
+      (element.title.toLowerCase()).includes(lowerCaseEvent))
+      || (element.description.toLowerCase()).includes(lowerCaseEvent));
+  };
```

Comment:
Do not do so ever)
Move visibleMovies into react component and filtrate it on render

Reply:
fixed)